### PR TITLE
LPS-127092 Locale Selector is missing under Documents And Media upload of a Document Type

### PR DESCRIPTION
**Pull request by boton**

### CODE REVIEW

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:forward

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by boton** [comment]
ci:forward

**Comment by boton** [comment]
ci:stop

**Comment by boton** [comment]
ci:forward

**Comment by boton** [comment]
ci:stop

**Comment by boton** [comment]
@4lejandrito we need an initial value in [the input](https://github.com/liferay-lima/liferay-portal/pull/1451/commits/39759ad2701f91b5a307dbb456ec94c5bda625ea#diff-53625c038f1180da5fe874b5e3312bf12ec0a7edc9bb0f3e553887934807a9e0R384).

**Comment by boton** [comment]
This PR is on top of #1465 we need this fix for working.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		initialSelectedLanguageId
+	);
+
+	const handleLocaleChange = (localeId) => {
+		setSelectedLanguageId(localeId);
+		addAviableLocale(localeId, portletNamespace);
+	};
+
+	useEffect(() => {
+		ddmStructureIds.forEach((ddmStructureId) => {
```
Comment (carloslancha) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js:111]: how does this works? Is the `dataEngineLayoutRenderer` component instantiated every time `selectedLanguageId` changes? Sounds weird if so. In this case we're running this code every time `portletNamespace` ,  `selectedLanguageId` and `ddmStructureIds` changes, but I guess we only want to run it if `ddmStrucureIds` change.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+LanguageSelector.propTypes = {
+	languageIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+	onChange: PropTypes.func.isRequired,
+	selectedLanguageId: PropTypes.string.isRequired,
+};
+
+function addAviableLocale(localeId, portletNamespace) {
+	const availableLocalesInput = document.getElementById(
```
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js:80]: jsp https://github.com/liferay-lima/liferay-portal/blob/74b8d86b3c23cf240a4d801e5cc115cf1bc51f21/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp#L384

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js (hunk 1 above)
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js:111]: No, we want to run if `selectedLanguageId ` changes, but the other two are props if some change we want to run too.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+							key={id}
+							onClick={() => {
+								onChange(id);
+								setIsDropdownOpen(false);
+							}}
+						>
+							<span className="inline-item inline-item-before">
+								<ClayIcon symbol={icon} />
+							</span>
+							{text}{' '}
```
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js:64]: I'm not sure it about buts looks like nothing or spacing the plan is to make it replaceable in the future for a common language selector component.

We use in [translations](https://github.com/liferay/liferay-portal/blob/master/modules/apps/translation/translation-web/src/main/resources/META-INF/resources/js/translate/LanguageSelector.js) and with some variations in [friendly_url_history](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/friendly_url_history/LanguageSelector.js).

I'm not sure how it's the best approach to abstract it. 

What do you think?

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js]: moved into react component

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js]: removed in both files

**Comment by boton** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+	return {
+		icon,
+		text,
+	};
+}
+
+function LanguageSelector({languageIds, onChange, selectedLanguageId}) {
+	const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+	const language = getLanguage(selectedLanguageId);
```
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/LanguageSelector.js:33]: I like selectedLanguage

**Comment by boton** [comment]
We have a bug with external video shortcuts :(

```
Caused by: com.liferay.document.library.kernel.exception.NoSuchFileEntryMetadataException: No DLFileEntryMetadata exists with the key {DDMStructureId=37922, fileVersionId=0}
        at com.liferay.portlet.documentlibrary.service.persistence.impl.DLFileEntryMetadataPersistenceImpl.findByD_F(DLFileEntryMetadataPersistenceImpl.java:2333)
        at com.liferay.portlet.documentlibrary.service.impl.DLFileEntryMetadataLocalServiceImpl.getFileEntryMetadata(DLFileEntryMetadataLocalServiceImpl.java:111)
        at sun.reflect.GeneratedMethodAccessor720.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
        at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
        at com.sun.proxy.$Proxy179.getFileEntryMetadata(Unknown Source)
        at com.liferay.document.library.kernel.service.DLFileEntryMetadataLocalServiceUtil.getFileEntryMetadata(DLFileEntryMetadataLocalServiceUtil.java:346)
        at com.liferay.document.library.web.internal.util.DDMStructureUtil.getTranslatedLanguageIds(DDMStructureUtil.java:76)
        at org.apache.jsp.document_005flibrary.edit_005ffile_005fentry_jsp._jspService(edit_005ffile_005fentry_jsp.java:2255)
        ... 233 more

```

**Review by carloslancha — APPROVED** [review]
LGTM

**Comment by 4lejandrito** [comment]
I'll fix the bug later today @boton.

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**